runtime/lua/vim/_editor.lua

```diff
@@ -670,28 +670,33 @@ local on_key_cbs = {} --- @type table<integer,function>
 ---          are applied, and {typed} is the key(s) before mappings are applied.
 ---          {typed} may be empty if {key} is produced by non-typed key(s) or by the
 ---          same typed key(s) that produced a previous {key}.
+---          If {opts.may_discard} then when {fn} returns
+---          true, {key} is ignored.
 ---          When {fn} is `nil` and {ns_id} is specified, the callback associated with
 ---          namespace {ns_id} is removed.
 ---@param ns_id integer? Namespace ID. If nil or 0, generates and returns a
 ---                      new |nvim_create_namespace()| id.
+---@param opts table|nil Optional parameters
+---              - may_discard {fn} controls discard (default false)
```

Comment:
Why is this option needed? Previously, the spec for `fn` was that its return type is "void". If we now treat a `true` return value specially, we don't need to make that optional.

Reply:
> Why is this option needed? Previously, the spec for `fn` was that its return type is "void". If we now treat a `true` return value specially, we don't need to make that optional.

It's not needed. It's there as a compatibility thing, even then it seemed a stretch. I'll remove it unless there's further comment to the contrary.